src/lib/engine.rs

```diff
@@ -60,7 +90,24 @@ fn apply_unique<'a>(cfg: &'a Config, repo: &dyn MailRepo) -> Result<HashMap<Path
             write!(query_str, " AND date:\"{}_days\"..", days)?;
         }
         debug!("using query: {}", query_str);
-        let messages = repo.search_message(&query_str)?;
+        let all_messages = repo.search_message(&query_str)?;
```

Comment:
If you extract this block of searching and filtering (lines 93-110), you can re-use it above in the check for overlap (it also makes writing test code easier).

Reply:
What do you mean ? In code I didn't push, I had implemented it as a method for the trait `MailRepo` (which used `MailRepo::search_message`, not by copying its code). But you said I should do it in the function instead.

Reply:
Sorry for the misunderstanding. To clarify, I meant to filter the messages  _outside_ the trait, i.e. without adding any new methods to it. The DRY principle still applies, so a simple approach would be to create a small private function and call it twice.

Reply:
I guess it's ok now.